eth2near/eth2near-block-relay/index.js

```diff
@@ -184,35 +196,42 @@ class Eth2NearRelay {
           for (let i = clientBlockNumber + 1; i <= endBlock; i++) {
             if (this.validateHeaderMode === 'bsc') {
               const block = await this.robustWeb3.getBlock(i)
+              if (await isBlockAlreadySubmitted(block)) {
+                if (endBlock <= chainBlockNumber) {
+                  endBlock++
+                  continue
+                }
+              }
               blockPromises.push({ header_rlp: block })
             } else {
               blockPromises.push(this.getParseBlock(i))
             }
           }
 
           const blocks = await Promise.all(blockPromises)
+          const firstBlock = blocks[0].header_rlp
           console.log(
-            `Got and parsed block ${clientBlockNumber + 1} to block ${endBlock}`
+            `Got and parsed block ${firstBlock.number} to block ${endBlock}`
           )
 
           // Send all transactions in a single batch, so they are processed in order.
           const actions = []
-          for (let i = clientBlockNumber + 1, j = 0; i <= endBlock; i++, j++) {
-            const action = this.submitBlock(blocks[j], i)
+          for (const block of blocks) {
+            const action = this.submitBlock(block)
             actions.push(action)
           }
 
           const task = this.ethClientContract.account.signAndSendTransaction(this.ethClientContract.contractId, actions)
 
           console.log(
-            `Submit txn to add block ${clientBlockNumber + 1
+            `Submit txn to add block ${firstBlock.number
             } to block ${endBlock}`
```

Comment:
This should be a single line for sure

Reply:
Fixed f3c20251c33750280ab677256c6c7f6da1f438fc